### PR TITLE
Task #5252 - User view and search books name, category, content

### PR DESCRIPTION
**Pull request by taylq**
https://edu-redmine.framgia.vn/issues/5252?issue_count=31&issue_position=8&next_issue_id=5250&prev_issue_id=5253

### CODE REVIEW

**Comment by taylq** [comment]
ready

**Comment by taylq** [comment]
ready

File: app/controllers/books_controller.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class BooksController < ApplicationController
+  def index
+    @books = Book.select_attr.page(params[:page]).per(Settings.book_per_page).search params[:search]
+    @categories = Category.all
```
Comment (huuphuc93) [inline app/controllers/books_controller.rb]: k dùng .all nhé

File: app/controllers/books_controller.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class BooksController < ApplicationController
+  def index
+    @books = Book.select_attr.page(params[:page]).per(Settings.book_per_page).search params[:search]
```
Comment (huuphuc93) [inline app/controllers/books_controller.rb]: quá 80 ký tự

File: app/models/book.rb [hunk 3]
```diff
@@ -7,4 +7,18 @@ class Book < ApplicationRecord
...
+  scope :select_attr, ->{select(:id, :category_id, :publisher_id, :name, :content, :number_of_page, :status, :image)}
+  scope :search_scope, ->(search){joins(:publisher).joins(book_authors: :author)
+    .where "books.name like '%#{search}%' or books.content like '%#{search}%' or publishers.name like '%#{search}%' or authors.name like '%#{search}%'"}
+
+  class << self
+    def search search
+      if search
+        search_scope search
+      else
+        Book.all
```
Comment (huuphuc93) [inline app/models/book.rb]: .all này

File: app/models/book.rb [hunk 4]
```diff
@@ -7,4 +7,18 @@ class Book < ApplicationRecord
 
   belongs_to :category
   belongs_to :publisher
+  default_scope -> {order(created_at: :desc)}
+  scope :select_attr, ->{select(:id, :category_id, :publisher_id, :name, :content, :number_of_page, :status, :image)}
+  scope :search_scope, ->(search){joins(:publisher).joins(book_authors: :author)
```
Comment (huuphuc93) [inline app/models/book.rb]: dùng do end thay vì {}

**Comment by huuphuc93** [comment]
commented

**Comment by taylq** [comment]
ready

**Pull request merged**